src/main/java/org/zalando/nakadi/service/CursorsService.java

```diff
@@ -159,14 +159,17 @@ private boolean processCursor(final String subscriptionId, final SubscriptionCur
                 cursorToProcess.getPartition(),
                 cursorToProcess.getOffset());
 
-        topicRepository.validateCommitCursor(toProcess);
-        return commitCursor(subscriptionId, eventType.getTopic(), cursorToProcess);
+        timelineService.getTopicRepository(eventType).validateCommitCursor(toProcess);
+        return commitCursor(subscriptionId, eventType, cursorToProcess);
     }
 
-    private boolean commitCursor(final String subscriptionId, final String eventType, final SubscriptionCursor cursor)
+    private boolean commitCursor(final String subscriptionId,
+                                 final EventType eventType,
+                                 final SubscriptionCursor cursor)
             throws ServiceUnavailableException, NoSuchSubscriptionException, InvalidCursorException {
 
-        final String offsetPath = format(PATH_ZK_OFFSET, subscriptionId, eventType, cursor.getPartition());
+        final String topic = eventType.getTopic();
```

Comment:
This is something that maybe doesn't make sense anymore. Here we can see that `event type` is coupled to the `kafka topic name`. But since this relation is made through a timeline now, it seems that the proper way to get this topic is through `event type + cursors` from the timelines.